### PR TITLE
Packaging, pre-commit and CLI

### PR DESCRIPTION
**Pull request by eggplants**
```
$ pip install .
$ ghst -h  # or: github-stars-by-topic -h
usage: ghst [-h] [-t ID] [-u ID] [-p PASSWORD] [-V]

Generate a list of your GitHub stars by topic - automatically!

optional arguments:
  -h, --help             show this help message and exit
  -t ID, --target-username ID
                         gh username to search (default: None)
  -u ID, --username ID   gh username to login (default: None)
  -p PASSWORD, --password PASSWORD
                         gh password to login (default: None)
  -V, --version          show program's version number and exit

$ ghst
Your Github Username: ...
```

---

If you are ok, please:
- change Workflow permissions into read/write
- add PyPI token as `PYPI_PASSWORD` to secrets
- run `git tag vX.X.X && git push --tags`

It enables to publish this project on pypi+releases+ghcr.

### CODE REVIEW

**Comment by lorey** [comment]
Thanks so much for this. I sadly haven't been able to review it yet, but amazing how much work you put into this! This has come a long way from the quick proof of concept I published years ago. Thanks!

File: example/tj/ruby-bindings-sinatra/README.md [hunk 1]
```diff
@@ -175,7 +175,7 @@ also defined by the following keywords: syntax, framework, session, rack, manage
 - [samg/timetrap](https://github.com/samg/timetrap)
   Simple command line timetracker
 - [leereilly/games](https://github.com/leereilly/games)
-  :video_game: A list of popular/awesome videos games, add-ons, maps, etc. hosted on GitHub. Any genre. Any platform. Any engine.
+  :video_game: A list of popular/awesome videos games, add-owns, maps, etc. hosted on GitHub. Any genre. Any platform. Any engine.
```
Comment (eggplants) [inline example/tj/ruby-bindings-sinatra/README.md:178]: mis-formatting by typos

File: example/tj/value-object-array/README.md [hunk 2]
```diff
@@ -75,7 +75,7 @@ also defined by the following keywords: return, true, false, returns, values, js
 - [coolony/kiwi](https://github.com/coolony/kiwi)
   Simple yet powerful asynchronous JavaScript template engine based on jQuery Template syntax, usable server-side or client-side.
 - [hashicorp/consul-template](https://github.com/hashicorp/consul-template)
-  Template rendering, notifier, and supervisor for @HashiCorp Consul and Vault data
+  Template rendering, notifier, and supervisor for @HashCorp Consul and Vault data
```
Comment (eggplants) [inline example/tj/value-object-array/README.md:78]: mis-formatting

File: example/lorey/chrome-firefox-browser/README.md [hunk 3]
```diff
@@ -7,7 +7,7 @@ also defined by the following keywords: extension, webkit, extensions, custom, c
 - [Langhalsdino/Amazon-Smile-Redirect-Chrome-Extension](https://github.com/Langhalsdino/Amazon-Smile-Redirect-Chrome-Extension)
   Amazon.de smiles redirect plugin for Chrome
 - [mdn/webextensions-examples](https://github.com/mdn/webextensions-examples)
-  Example Firefox add-ons created using the WebExtensions API
+  Example Firefox add-owns created using the WebExtensions API
```
Comment (eggplants) [inline example/lorey/chrome-firefox-browser/README.md:10]: mis-formatting

File: example/lorey/public-gnu-json/README.md [hunk 4]
```diff
@@ -51,7 +51,7 @@ also defined by the following keywords: core, cldr, general, terms, apis, non, d
 - [kamranahmedse/developer-roadmap](https://github.com/kamranahmedse/developer-roadmap)
   Roadmap to becoming a web developer in 2017
 - [mdn/webextensions-examples](https://github.com/mdn/webextensions-examples)
-  Example Firefox add-ons created using the WebExtensions API
+  Example Firefox add-owns created using the WebExtensions API
```
Comment (eggplants) [inline example/lorey/public-gnu-json/README.md:54]: mis-formatting

File: example/tj/aws-amazon-s3/README.md [hunk 5]
```diff
@@ -93,7 +93,7 @@ also defined by the following keywords: service, deploy, ec2, credentials, key,
 - [hashicorp/envconsul](https://github.com/hashicorp/envconsul)
   Read and set environmental variables for processes from Consul.
 - [hashicorp/consul-template](https://github.com/hashicorp/consul-template)
-  Template rendering, notifier, and supervisor for @HashiCorp Consul and Vault data
+  Template rendering, notifier, and supervisor for @HashCorp Consul and Vault data
```
Comment (eggplants) [inline example/tj/aws-amazon-s3/README.md:96]: mis-formatting

File: example/tj/browser-script-web/README.md [hunk 6]
```diff
@@ -251,7 +251,7 @@ also defined by the following keywords: cross, browsers, ui, chrome, style, dom,
 - [jwagner/smartcrop.js](https://github.com/jwagner/smartcrop.js)
   Content aware image cropping
 - [leereilly/games](https://github.com/leereilly/games)
-  :video_game: A list of popular/awesome videos games, add-ons, maps, etc. hosted on GitHub. Any genre. Any platform. Any engine.
+  :video_game: A list of popular/awesome videos games, add-owns, maps, etc. hosted on GitHub. Any genre. Any platform. Any engine.
```
Comment (eggplants) [inline example/tj/browser-script-web/README.md:254]: mis-formatting

File: example/tj/canvas-image-color/README.md [hunk 7]
```diff
@@ -243,7 +243,7 @@ also defined by the following keywords: width, height, png, images, svg, rgb, re
 - [chrislgarry/Apollo-11](https://github.com/chrislgarry/Apollo-11)
   Original Apollo 11 Guidance Computer (AGC) source code for the command and lunar modules.
 - [leereilly/games](https://github.com/leereilly/games)
-  :video_game: A list of popular/awesome videos games, add-ons, maps, etc. hosted on GitHub. Any genre. Any platform. Any engine.
+  :video_game: A list of popular/awesome videos games, add-owns, maps, etc. hosted on GitHub. Any genre. Any platform. Any engine.
```
Comment (eggplants) [inline example/tj/canvas-image-color/README.md:246]: mis-formatting

File: example/tj/client-redis-cluster/README.md [hunk 8]
```diff
@@ -261,13 +261,13 @@ also defined by the following keywords: store, irc, powerful, key, commands, rep
 - [c4milo/handlers](https://github.com/c4milo/handlers)
   Go's HTTP handlers I use in my projects
 - [hashicorp/consul-template](https://github.com/hashicorp/consul-template)
-  Template rendering, notifier, and supervisor for @HashiCorp Consul and Vault data
+  Template rendering, notifier, and supervisor for @HashCorp Consul and Vault data
```
Comment (eggplants) [inline example/tj/client-redis-cluster/README.md:264]: mis-formatting

File: example/tj/client-redis-cluster/README.md [hunk 9]
```diff
@@ -228,7 +228,7 @@ also defined by the following keywords: store, irc, powerful, key, commands, rep
 - [fogleman/Craft](https://github.com/fogleman/Craft)
   A simple Minecraft clone written in C using modern OpenGL (shaders).
 - [leereilly/games](https://github.com/leereilly/games)
-  :video_game: A list of popular/awesome videos games, add-ons, maps, etc. hosted on GitHub. Any genre. Any platform. Any engine.
+  :video_game: A list of popular/awesome videos games, add-owns, maps, etc. hosted on GitHub. Any genre. Any platform. Any engine.
```
Comment (eggplants) [inline example/tj/client-redis-cluster/README.md:231]: mis-formatting

File: example/tj/docker-container-image/README.md [hunk 10]
```diff
@@ -145,7 +145,7 @@ also defined by the following keywords: containers, registry, images, latest, ta
 - [scylladb/seastar](https://github.com/scylladb/seastar)
   High performance server-side application framework
 - [hashicorp/consul-template](https://github.com/hashicorp/consul-template)
-  Template rendering, notifier, and supervisor for @HashiCorp Consul and Vault data
+  Template rendering, notifier, and supervisor for @HashCorp Consul and Vault data
```
Comment (eggplants) [inline example/tj/docker-container-image/README.md:148]: mis-formatting

File: example/tj/err-func-fmt/README.md [hunk 11]
```diff
@@ -403,7 +403,7 @@ also defined by the following keywords: nil, main, import, println, error, print
 - [tsenart/vegeta](https://github.com/tsenart/vegeta)
   HTTP load testing tool and library. It's over 9000!
 - [hashicorp/consul-template](https://github.com/hashicorp/consul-template)
-  Template rendering, notifier, and supervisor for @HashiCorp Consul and Vault data
+  Template rendering, notifier, and supervisor for @HashCorp Consul and Vault data
```
Comment (eggplants) [inline example/tj/err-func-fmt/README.md:406]: mis-formatting

File: example/tj/fast-compression-performance/README.md [hunk 12]
```diff
@@ -314,7 +314,7 @@ also defined by the following keywords: memory, byte, encoding, op, faster, inte
 - [gocql/gocql](https://github.com/gocql/gocql)
   Package gocql implements a fast and robust Cassandra client for the Go programming language.
 - [leereilly/games](https://github.com/leereilly/games)
-  :video_game: A list of popular/awesome videos games, add-ons, maps, etc. hosted on GitHub. Any genre. Any platform. Any engine.
+  :video_game: A list of popular/awesome videos games, add-owns, maps, etc. hosted on GitHub. Any genre. Any platform. Any engine.
```
Comment (eggplants) [inline example/tj/fast-compression-performance/README.md:317]: mis-formatting

File: example/tj/git-clone-sh/README.md [hunk 13]
```diff
@@ -586,7 +586,7 @@ also defined by the following keywords: shell, cd, bash, repo, directory, commit
 - [c4milo/github-release](https://github.com/c4milo/github-release)
   Simple Github release command line tool
 - [hashicorp/consul-template](https://github.com/hashicorp/consul-template)
-  Template rendering, notifier, and supervisor for @HashiCorp Consul and Vault data
+  Template rendering, notifier, and supervisor for @HashCorp Consul and Vault data
```
Comment (eggplants) [inline example/tj/git-clone-sh/README.md:589]: mis-formatting

File: example/tj/implementation-language-hash/README.md [hunk 14]
```diff
@@ -202,7 +202,7 @@ also defined by the following keywords: functions, programming, pure, compiler,
 - [mruby/mruby](https://github.com/mruby/mruby)
   Lightweight Ruby
 - [leereilly/games](https://github.com/leereilly/games)
-  :video_game: A list of popular/awesome videos games, add-ons, maps, etc. hosted on GitHub. Any genre. Any platform. Any engine.
+  :video_game: A list of popular/awesome videos games, add-owns, maps, etc. hosted on GitHub. Any genre. Any platform. Any engine.
```
Comment (eggplants) [inline example/tj/implementation-language-hash/README.md:205]: mis-formatting

File: example/tj/parser-parsing-parse/README.md [hunk 15]
```diff
@@ -133,7 +133,7 @@ also defined by the following keywords: fast, streaming, json, agent, compiler,
 - [kangax/fabric.js](https://github.com/kangax/fabric.js)
   Javascript Canvas Library, SVG-to-Canvas (& canvas-to-SVG) Parser
 - [hashicorp/hcl](https://github.com/hashicorp/hcl)
-  HCL is the HashiCorp configuration language.
+  HCL is the HashCorp configuration language.
```
Comment (eggplants) [inline example/tj/parser-parsing-parse/README.md:136]: mis-formatting

File: example/tj/plugin-sketch-plugins/README.md [hunk 16]
```diff
@@ -238,6 +238,6 @@ also defined by the following keywords: jquery, elements, text, select, folder,
 - [kkga/spacegray](https://github.com/kkga/spacegray)
   A Hyperminimal UI Theme for Sublime Text
 - [hashicorp/consul-template](https://github.com/hashicorp/consul-template)
-  Template rendering, notifier, and supervisor for @HashiCorp Consul and Vault data
+  Template rendering, notifier, and supervisor for @HashCorp Consul and Vault data
```
Comment (eggplants) [inline example/tj/plugin-sketch-plugins/README.md:241]: mis-formatting

File: example/tj/port-tcp-host/README.md [hunk 17]
```diff
@@ -163,7 +163,7 @@ also defined by the following keywords: request, connections, send, response, lo
 - [boot2docker/boot2docker](https://github.com/boot2docker/boot2docker)
   Lightweight Linux for Docker
 - [hashicorp/consul-template](https://github.com/hashicorp/consul-template)
-  Template rendering, notifier, and supervisor for @HashiCorp Consul and Vault data
+  Template rendering, notifier, and supervisor for @HashCorp Consul and Vault data
```
Comment (eggplants) [inline example/tj/port-tcp-host/README.md:166]: mis-formatting

File: example/tj/ruby-bindings-sinatra/README.md [hunk 18]
```diff
@@ -137,7 +137,7 @@ also defined by the following keywords: syntax, framework, session, rack, manage
 - [markbates/genosaurus](https://github.com/markbates/genosaurus)
   A simple and easy to use generator system for Ruby
 - [hashicorp/hcl](https://github.com/hashicorp/hcl)
-  HCL is the HashiCorp configuration language.
+  HCL is the HashCorp configuration language.
```
Comment (eggplants) [inline example/tj/ruby-bindings-sinatra/README.md:140]: mis-formatting

File: example/tj/value-object-array/README.md [hunk 19]
```diff
@@ -173,7 +173,7 @@ also defined by the following keywords: return, true, false, returns, values, js
 - [pillarjs/cookies](https://github.com/pillarjs/cookies)
   Signed and unsigned cookies based on Keygrip
 - [hashicorp/hcl](https://github.com/hashicorp/hcl)
-  HCL is the HashiCorp configuration language.
+  HCL is the HashCorp configuration language.
```
Comment (eggplants) [inline example/tj/value-object-array/README.md:176]: mis-formatting

File: example/tj/web-development-contributing/README.md [hunk 20]
```diff
@@ -125,7 +125,7 @@ also defined by the following keywords: issues, issue, pull, repository, org, we
 - [inspircd/inspircd](https://github.com/inspircd/inspircd)
   A modular C++ IRC daemon (ircd).
 - [leereilly/games](https://github.com/leereilly/games)
-  :video_game: A list of popular/awesome videos games, add-ons, maps, etc. hosted on GitHub. Any genre. Any platform. Any engine.
+  :video_game: A list of popular/awesome videos games, add-owns, maps, etc. hosted on GitHub. Any genre. Any platform. Any engine.
```
Comment (eggplants) [inline example/tj/web-development-contributing/README.md:128]: mis-formatting

File: example/tj/web-development-contributing/README.md [hunk 21]
```diff
@@ -1189,7 +1189,7 @@ also defined by the following keywords: issues, issue, pull, repository, org, we
 - [tinylib/msgp](https://github.com/tinylib/msgp)
   A Go code generator for MessagePack / msgpack.org[Go]
 - [hashicorp/hcl](https://github.com/hashicorp/hcl)
-  HCL is the HashiCorp configuration language.
+  HCL is the HashCorp configuration language.
```
Comment (eggplants) [inline example/tj/web-development-contributing/README.md:1192]: mis-formatting

File: example/tj/windows-linux-os/README.md [hunk 22]
```diff
@@ -266,7 +266,7 @@ also defined by the following keywords: platform, python, mac, tool, program, co
 - [donpark/html2jade](https://github.com/donpark/html2jade)
   Converts HTML to Jade template. Not perfect but useful enough for non-daily conversions.
 - [leereilly/games](https://github.com/leereilly/games)
-  :video_game: A list of popular/awesome videos games, add-ons, maps, etc. hosted on GitHub. Any genre. Any platform. Any engine.
+  :video_game: A list of popular/awesome videos games, add-owns, maps, etc. hosted on GitHub. Any genre. Any platform. Any engine.
```
Comment (eggplants) [inline example/tj/windows-linux-os/README.md:269]: mis-formatting

**Comment by eggplants** [comment]
I have to fix mis-formatting:

- `add-owns` -> `add-ons`
- `HashCorp` -> `HashiCorp`

**Pull request closed without merging**